### DISCUSSION TITLE
Async storage returning null , when running through visionOs

### DISCUSSION THREAD

**Discussion by hafisaq** [callstack/react-native-visionos#145] [Q&A] [answered]
My current react native application uses mmkv and async storage. It is building and running properly if I build it through iOS (visionPro (designed for iphone)). But when I try to run it from visionOs im getting error of :

`Invariant Violation: "Skiply" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes
 BUNDLE  ./index.js 

 ERROR  Error: [@RNC/AsyncStorage]: NativeModule: AsyncStorage is null.

To fix this issue try these steps:

  • Uninstall, rebuild and restart the app.

  • Run the packager with `--reset-cache` flag.

  • If you are using CocoaPods on iOS, run `pod install` in the `ios` directory, then rebuild and re-run the app.

  • Make sure your project's `package.json` depends on `@react-native-async-storage/async-storage`, even if you only depend on it indirectly through other dependencies. CLI only autolinks native modules found in your `package.json`.

  • If this happens while testing with Jest, check out how to integrate AsyncStorage here: https://react-native-async-storage.github.io/async-storage/docs/advanced/jest

If none of these fix the issue, please open an issue on the GitHub repository: https://github.com/react-native-async-storage/async-storage/issues`

when I try removing the package it bypasses but MMKV returns an error :

` Error: Failed to create a new MMKV instance: The native MMKV Module could not be found.
* Make sure react-native-mmkv is correctly autolinked (run `npx react-native config` to verify)
* Make sure you ran `pod install` in the ios/ directory.
* Make sure you rebuilt the app., js engine: hermes
 LOG  Running "Skiply" with {"rootTag":121,"initialProps":{}}
 ERROR  Invariant Violation: "Skiply" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes`

**Comment by okwasniewski** [comment, marked as answer]
Hey, 

It looks like visionOS support was introduced to `@react-native-async-storage/async-storage` with this PR: https://github.com/react-native-async-storage/async-storage/pull/1063 Can you check if you are running the latest version? 

Regarding MMKV, it looks like it doesn't have official visionOS support.
